python/src/llvm.cc

```diff
@@ -377,6 +379,26 @@ void init_triton_llvm(py::module &&m) {
               fpm.addPass(BreakStructPhiNodesPass());
               fpm.addPass(InstCombinePass());
             });
+        bool enableAddressSanitizer =
+            mlir::triton::tools::getBoolEnv("TRITON_ENABLE_ADDRESS_SANITIZER");
+        if (enableAddressSanitizer) {
+          llvm::SmallVector<StringRef, 3> split;
+          StringRef(mod->getTargetTriple()).split(split, '-');
+          if (split[0] != "amdgcn") {
```

Comment:
Hmm, this is too AMD specific that I think it's better to refactor out. For example, what about having an additional parameter `enable_asan` to control this? We can turn it on only for AMD in the AMD backend.

Reply:
OK. Let me think about this. I think we still want to hard error out if in NV system and not silently just not enable it.

Reply:
Updated